src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs

```diff
@@ -35,4 +35,38 @@ static PageHandler CreatePageHandler(IPopup virtualView)
 			return (PageHandler)contentPage.ToHandler(mauiContext);
 		}
 	}
+
+	/// <summary>
+	/// Action that's triggered when the Popup is Closed.
+	/// </summary>
+	/// <param name="handler">An instance of <see cref="PopupHandler"/>.</param>
+	/// <param name="view">An instance of <see cref="IPopup"/>.</param>
+	/// <param name="result">We don't need to provide the result parameter here.</param>
+	public static void MapOnClosed(PopupHandler handler, IPopup view, object? result)
+	{
+		var parent = view.Parent as Element;
+		if (parent is not null)
+		{
+			var contentPage = parent.LogicalChildrenInternal
+				.Where(
+					x => x is ContentPage
+				)
+				.LastOrDefault();
```

Comment:
Any risk we'll end up removing something that isn't a popup's ContentPage here? 

Can we get a reference to the specific ContentPage using handler.PlatformView.Control?

Reply:
@NeilMalcolm , Thank you for your review.
In cases where multiple Popups are displayed, a problem may arise if the Popups are not closed in the reverse order of the display order. I'll look into another fix.

Reply:
@NeilMalcolm , I have made the following changes.
With this method, there is no need to consider the display order of Popups.

```
	public static void MapOnClosed(PopupHandler handler, IPopup view, object? result)
	{
		var parent = view.Parent as Element;
		if (parent is not null)
		{
			if (handler.VirtualView is Popup popup)
			{
				if (popup.Content is not null)
				{
					if (popup.Content.Parent is ContentPage contentPage)
					{
						parent.RemoveLogicalChild(contentPage);
					}
				}
				parent.RemoveLogicalChild(popup);
			}
		}
	}
```